event/src/test/java/com/bazaarvoice/emodb/event/db/astyanax/DefaultSlabAllocatorTest.java

```diff
@@ -0,0 +1,165 @@
+        } catch (Exception e) {
+            _log.error("ERROR: " + e.getClass().getName() + " thrown when " + (Constants.MAX_SLAB_SIZE*3/2) + " events allocated");
+            assert(false);
+        }
+    }
+
+    @Test
+    public void allocateMaxEvents() {
+        List<Integer> sizes = new ArrayList<Integer>();
+        for (int i=0; i<Constants.MAX_SLAB_SIZE; i++) {
+            sizes.add(64);
+        }
+        DefaultSlabAllocator slabAllocator = new DefaultSlabAllocator(mock(LifeCycleRegistry.class), mock(ManifestPersister.class), "metrics", mock(MetricRegistry.class));
+        try {
+            slabAllocator.allocate("mychannel", Constants.MAX_SLAB_SIZE, Iterators.peekingIterator(sizes.iterator()));
+            _log.info("SUCCESS: No exception thrown when " + Constants.MAX_SLAB_SIZE + " events allocated");
+        } catch (Exception e) {
+            _log.error("ERROR: " + e.getClass().getName() + " thrown when " + Constants.MAX_SLAB_SIZE + " events allocated");
+            assert(false);
+        }
+    }
+
+    @Test
+    public void allocateFewerThanMaxEvents() {
+        List<Integer> sizes = new ArrayList<Integer>();
+        for (int i=0; i<Constants.MAX_SLAB_SIZE/2; i++) {
+            sizes.add(64);
+        }
+        DefaultSlabAllocator slabAllocator = new DefaultSlabAllocator(mock(LifeCycleRegistry.class), mock(ManifestPersister.class), "metrics", mock(MetricRegistry.class));
+        try {
+            slabAllocator.allocate("mychannel", Constants.MAX_SLAB_SIZE/2, Iterators.peekingIterator(sizes.iterator()));
+            _log.info("SUCCESS: No exception thrown when " + (Constants.MAX_SLAB_SIZE/2) + " events allocated");
+        } catch (Exception e) {
+            _log.error("ERROR: " + e.getClass().getName() + " thrown when " + (Constants.MAX_SLAB_SIZE/2) + " events allocated");
+            assert(false);
+        }
+    }
+
+
+    @Test
+    public void computeAllocationGreaterThanMaxEvents() {
+        List<Integer> sizes = new ArrayList<Integer>();
+        for (int i=0; i<Constants.MAX_SLAB_SIZE*3/2; i++) {
+            sizes.add(64);
+        }
+        try {
+            Pair<Integer, Integer> allocation = DefaultSlabAllocator.defaultAllocationCount(0,0, Iterators.peekingIterator(sizes.iterator()));
+            assertTrue(allocation.getLeft() == Constants.MAX_SLAB_SIZE, "ERROR: allocation has " + allocation.getLeft() + " slots, " + Constants.MAX_SLAB_SIZE + " expected");
+            _log.info("SUCCESS: allocation has " + allocation.getLeft() + " slots, " + Constants.MAX_SLAB_SIZE + " expected when " + 0 + " slots already used and " + (Constants.MAX_SLAB_SIZE*3/2) + " requested");
+        } catch (Exception e) {
+            _log.error("ERROR: " + e.getClass().getName() + " thrown when " + (Constants.MAX_SLAB_SIZE*3/2) + " events allocated");
+            assert(false);
+        }
+    }
+
+    @Test
+    public void computeAllocationMaxEvents() {
+        List<Integer> sizes = new ArrayList<Integer>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.qa.bazaarvoice.com/coding_rules#rule_key=squid%3AS2293)